net/mwan3/files/lib/mwan3/mwan3.sh

```diff
@@ -929,92 +917,92 @@ mwan3_set_user_iptables_rule()
-						$IPT -N mwan3_rule_$1
+					if ! $IPT -S "mwan3_rule_$1" &> /dev/null; then
+						$IPT -N "mwan3_rule_$1"
 					fi
 
-					$IPT -F mwan3_rule_$1
+					$IPT -F "mwan3_rule_$1"
 				done
 
-				$IPS -! create mwan3_sticky_v4_$rule \
-					hash:ip,mark markmask $MMX_MASK \
-					timeout $timeout
-				$IPS -! create mwan3_sticky_v6_$rule \
-					hash:ip,mark markmask $MMX_MASK \
-					timeout $timeout family inet6
-				$IPS -! create mwan3_sticky_$rule list:set
-				$IPS -! add mwan3_sticky_$rule mwan3_sticky_v4_$rule
-				$IPS -! add mwan3_sticky_$rule mwan3_sticky_v6_$rule
+				$IPS -! create "mwan3_sticky_v4_$rule" \
+					hash:ip,mark markmask "$MMX_MASK" \
+					timeout "$timeout"
+				$IPS -! create "mwan3_sticky_v6_$rule" \
+					hash:ip,mark markmask "$MMX_MASK" \
+					timeout "$timeout" family inet6
+				$IPS -! create "mwan3_sticky_$rule" list:set
+				$IPS -! add "mwan3_sticky_$rule" "mwan3_sticky_v4_$rule"
+				$IPS -! add "mwan3_sticky_$rule" "mwan3_sticky_v6_$rule"
 
 				config_foreach mwan3_set_sticky_iptables interface
 
 				for IPT in "$IPT4" "$IPT6"; do
-					$IPT -A mwan3_rule_$1 \
+					$IPT -A "mwan3_rule_$1" \
 						-m mark --mark 0/$MMX_MASK \
-						-j $policy
-					$IPT -A mwan3_rule_$1 \
+						-j "$policy"
+					$IPT -A "mwan3_rule_$1" \
 						-m mark ! --mark 0xfc00/0xfc00 \
-						-j SET --del-set mwan3_sticky_$rule src,src
-					$IPT -A mwan3_rule_$1 \
+						-j SET --del-set "mwan3_sticky_$rule" src,src
+					$IPT -A "mwan3_rule_$1" \
 						-m mark ! --mark 0xfc00/0xfc00 \
-						-j SET --add-set mwan3_sticky_$rule src,src
+						-j SET --add-set "mwan3_sticky_$rule" src,src
 				done
 
 				policy="mwan3_rule_$1"
 			else
 				policy="mwan3_policy_$use_policy"
 
 				for IPT in "$IPT4" "$IPT6"; do
-					if ! $IPT -S $policy &> /dev/null; then
-						$IPT -N $policy
+					if ! $IPT -S "$policy" &> /dev/null; then
+						$IPT -N "$policy"
 					fi
 				done
 
 			fi
 		fi
 		for IPT in "$IPT4" "$IPT6"; do
-			[ "$family" == "ipv4" ] && [ "$IPT" == "$IPT6" ] && continue
-			[ "$family" == "ipv6" ] && [ "$IPT" == "$IPT4" ] && continue
+			[ "$family" = "ipv4" ] && [ "$IPT" = "$IPT6" ] && continue
+			[ "$family" = "ipv6" ] && [ "$IPT" = "$IPT4" ] && continue
 			[ "$global_logging" = "1" ] && [ "$rule_logging" = "1" ] && {
 				$IPT -A mwan3_rules \
-				     -p $proto \
-				     ${src_ip:+-s} $src_ip \
-				     ${src_dev:+-i} $src_dev \
-				     ${dest_ip:+-d} $dest_ip\
+				     -p "$proto" \
+				     ${src_ip:+-s} "$src_ip" \
+				     ${src_dev:+-i} "$src_dev" \
+				     ${dest_ip:+-d} "$dest_ip" \
 				     $ipset \
-				     ${src_port:+-m} ${src_port:+multiport} ${src_port:+--sports} $src_port \
-				     ${dest_port:+-m} ${dest_port:+multiport} ${dest_port:+--dports} $dest_port \
+				     ${src_port:+-m} ${src_port:+multiport} ${src_port:+--sports} "$src_port" \
+				     ${dest_port:+-m} ${dest_port:+multiport} ${dest_port:+--dports} "$dest_port" \
```

Comment:
This appears to break my configuration. If the src_port, etc,  is not set you end up with extra blank arguments that iptables does not like

Reply:
Thanks for pointing that out, I'm gonna fix that right now.
So we can merge my pullrequest and then you could [rebase](https://github.com/openwrt/packages/pull/12814) yours.